src/components/sections/contest/GithubFileSelect.tsx

```diff
@@ -0,0 +1,122 @@
+'use client'
+
+import {useState} from 'react'
+import {ArrowLeft} from 'lucide-react'
+import {signIn, useSession} from 'next-auth/react'
+
+import {Button} from '@/components/ui/Button'
+import {
+  useGetPublicRepos,
+  useGetRepoFiles,
+} from '@/lib/queries/github/getGitHub'
+import {
+  GetPublicReposResponse,
+  GetRepoFilesParams,
+  GetRepoFilesResponse,
+} from '@/server/actions/github/getGithub'
+import {Checkbox} from '@/components/ui/Checkbox'
+import {ScrollArea} from '@/components/ui/ScrollArea'
+
+const GithubFileSelect = () => {
+  const session = useSession()
+
+  const [selectedRepo, setSelectedRepo] = useState<GetRepoFilesParams | null>(
+    null,
+  )
+  const {data: publicReposData, isLoading: publicReposIsLoading} =
+    useGetPublicRepos()
+  const {data: repoFilesData, isLoading: repoFilesIsLoading} = useGetRepoFiles(
```

Comment:
These hooks should be enabled only when oauth provider is github, otherwise they're failing with errors.

Reply:
Good catch